kodilla-stream/src/test/java/com/kodilla/stream/portfolio/BoardTestSuite.java

```diff
@@ -0,0 +1,173 @@
+        //board
+        Board project = new Board("Project Weather Prediction");
+        project.addTaskList(taskListToDo);
+        project.addTaskList(taskListInProgress);
+        project.addTaskList(taskListDone);
+        return project;
+    }
+
+    @Test
+    public void testAddTaskList(){
+        //Given
+        Board project = prepareTestData();
+        //When
+
+        //Then
+        Assert.assertEquals(3,project.getTaskLists().size());
+    }
+
+    @Test
+    public void testAddTaskListFindUsersTasks(){
+        //Given
+        Board project = prepareTestData();
+        User user = new User("developer1", "John Smith");
+        //When
+        List<Task> tasks = project.getTaskLists().stream()
+                .flatMap(l -> l.getTasks().stream())
+                .filter(t -> t.getAssignedUser().equals(user))
+                .collect(toList());
+        //Then
+        Assert.assertEquals(2, tasks.size());
+        Assert.assertEquals(user,tasks.get(0).getAssignedUser());
+        Assert.assertEquals(user, tasks.get(1).getAssignedUser());
+    }
+
+    @Test
+    public void testAddTaskListFindOutdatedTasks(){
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        List<TaskList> undoneTask = new ArrayList<>();
+        undoneTask.add(new TaskList("To do"));
+        undoneTask.add(new TaskList("In progress"));
+        List<Task> tasks = project.getTaskLists().stream()
+                .filter(undoneTask::contains)
+                .flatMap(tl -> tl.getTasks().stream())
+                .filter(t -> t.getDeadline().isBefore(LocalDate.now()))
+                .collect(toList());
+        //Then
+        Assert.assertEquals(1, tasks.size());
+        Assert.assertEquals("HQLs for analysis", tasks.get(0).getTitle());
+
+    }
+
+    @Test
+    public void testAddTaskListFindLongTasks(){
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        List<TaskList> inProgressTasks = new ArrayList<>();
+        inProgressTasks.add(new TaskList("In progress"));
+        long longTask = project.getTaskLists().stream()
+                .filter(inProgressTasks::contains)
+                .flatMap(tl -> tl.getTasks().stream())
+                .map(t -> t.getCreated())
+                .filter(d -> d.compareTo(LocalDate.now().minusDays(10)) <= 0)
+                .count();
+
+        //Then
+        Assert.assertEquals(2, longTask);
+    }
+
+    @Test
+    public void testAddTaskListAverageWorkingOnTask(){
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        List<TaskList> inProgressTasks = new ArrayList<>();
+        inProgressTasks.add(new TaskList("In progress"));
+        long longTask = (long) project.getTaskLists().stream()
+                .filter(inProgressTasks::contains)
+                .flatMap(tl -> tl.getTasks().stream())
+                .mapToInt(n -> n.getCreated().compareTo(LocalDate.now()))
```

Comment:
poprawne to 
` .mapToInt(n -> (int)(LocalDate.now().toEpochDay() - n.getCreated().toEpochDay()))`

wynik przecieciez nie moze byc ujemny..

compareTo - tutaj po wejsciu w description metody jest odp. Porownywe sa dni jesli miesiac jest ten sam. Miesiace jesli rok jest ten sam. Dalej rok w kolejnosci.

Reply:
* dobrze ze wylapales blad z > -10